query-service-impl/src/test/java/org/hypertrace/core/query/service/postgres/QueryRequestToPostgresSQLConverterTest.java

```diff
@@ -0,0 +1,1212 @@
+            + "' "
+            + "AND span_name IN ('1'') OR tenant_id = ''tenant2'' and span_name IN (''1')",
+        tableDefinition,
+        executionContext);
+  }
+
+  @Test
+  void testQueryWithLikeOperator() {
+    Builder builder = QueryRequest.newBuilder();
+    Expression spanId = createColumnExpression("Span.id").build();
+    builder.addSelection(spanId);
+
+    Filter likeFilter =
+        Filter.newBuilder()
+            .setOperator(Operator.LIKE)
+            .setLhs(spanId)
+            .setRhs(createStringLiteralValueExpression("042e5523ff6b2506"))
+            .build();
+    builder.setFilter(likeFilter);
+
+    TableDefinition tableDefinition = getDefaultTableDefinition();
+    defaultMockingForExecutionContext();
+
+    assertSQLQuery(
+        builder.build(),
+        "SELECT encode(span_id, 'hex') FROM public.\"span-event-view\" "
+            + "WHERE "
+            + tableDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "' "
+            + "AND span_id like decode('042e5523ff6b2506', 'hex')",
+        tableDefinition,
+        executionContext);
+  }
+
+  @Test
+  void testQueryWithContainsKeyOperator() {
+    Builder builder = QueryRequest.newBuilder();
+    builder.addSelection(createColumnExpression("Span.tags"));
+    builder.setFilter(createContainsKeyFilter("Span.tags", "FLAGS"));
+
+    TableDefinition tableDefinition = getDefaultTableDefinition();
+    defaultMockingForExecutionContext();
+
+    assertSQLQuery(
+        builder.build(),
+        "SELECT cast(tags as text) FROM public.\"span-event-view\" "
+            + "WHERE "
+            + tableDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "' "
+            + "AND tags->>'flags' IS NOT NULL",
+        tableDefinition,
+        executionContext);
+  }
+
+  @Test
+  void testQueryWithNotContainsKeyOperator() {
+    Builder builder = QueryRequest.newBuilder();
+    builder.addSelection(createColumnExpression("Span.tags"));
+    builder.setFilter(createNotContainsKeyFilter("Span.tags", "FLAGS"));
+
+    TableDefinition tableDefinition = getDefaultTableDefinition();
+    defaultMockingForExecutionContext();
+
+    assertSQLQuery(
+        builder.build(),
+        "SELECT cast(tags as text) FROM public.\"span-event-view\" "
+            + "WHERE "
+            + tableDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "' "
+            + "AND tags->>'flags' IS NULL",
+        tableDefinition,
+        executionContext);
+  }
+
+  @Test
+  void testQueryWithContainsKeyValueOperator() {
+    Builder builder = QueryRequest.newBuilder();
+    Expression spanTag = createColumnExpression("Span.tags").build();
+    builder.addSelection(spanTag);
+
+    Expression tag = createStringArrayLiteralValueExpression(List.of("FLAGS", "0"));
+    Filter likeFilter =
+        Filter.newBuilder()
+            .setOperator(Operator.CONTAINS_KEYVALUE)
```

Comment:
Note: `CONTAINS_KEYVALUE` is deprecated.

Reply:
okay